src/transformers/models/flava/modeling_flava.py

```diff
@@ -0,0 +1,2101 @@
+            Indices should be in `[-100, 0, ..., text_config.vocab_size]` (see `input_ids` docstring) Tokens with
+            indices set to `-100` are ignored (masked), the loss is only computed for the tokens with labels n `[0,
+            ..., text_config.vocab_size]`
+
+        mim_labels (`torch.LongTensor` of shape `(batch_size, image_num_patches)`, *optional*):
+            Labels for computing the image and multimodal masked modeling loss. Indices should be in `[-100, 0, ...,
+            image_config.vocab_size]`. Tokens with indices set to `-100` are ignored (masked), the loss is only
+            computed for the tokens with labels n `[0, ..., image_config.vocab_size]`. See [`FLAVACodebook`] to
+            understand how to generate mim_labels.
+
+        itm_labels (`torch.LongTensor` of shape `(batch_size, 1)`, *optional*):
+            Labels for computing the image-text matching loss. 0 means the pairs don't match and 1 means they match.
+            The pairs with 0 will be skipped for calculation of MMM and global contrastive losses as well.
+
+        return_loss (`bool`, *optional*, default to None):
+            Whether to return calculated loss or not.
+"""
+    + FLAVA_INPUTS_DOCSTRING_COMMON
+)
+
+
+class FLAVAPreTrainedModel(PreTrainedModel):
+    """
+    An abstract class to handle weights initialization and a simple interface for downloading and loading pretrained
+    models.
+    """
+
+    config_class = FLAVAConfig
+    base_model_prefix = "flava"
+    supports_gradient_checkpointing = True
+
+    def _init_weights(self, module: Union[nn.Linear, nn.Conv2d, nn.LayerNorm]) -> None:
+        """Initialize the weights"""
+        if isinstance(module, (nn.Linear, nn.Conv2d)):
+            # Slightly different from the TF version which uses truncated_normal for initialization
+            # cf https://github.com/pytorch/pytorch/pull/5617
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, nn.Embedding):
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
+            if module.padding_idx is not None:
+                module.weight.data[module.padding_idx].zero_()
+        elif isinstance(module, nn.LayerNorm):
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+
+    def _set_gradient_checkpointing(self, module: FLAVAEncoder, value: bool = False) -> None:
+        if isinstance(module, FLAVAEncoder):
+            module.gradient_checkpointing = value
+
+
+@add_start_docstrings(
+    "The bare FLAVA Image Model transformer outputting raw hidden-states without any specific head on top.",
+    FLAVA_START_DOCSTRING.format(config="FLAVAImageConfig"),
+)
+class FLAVAImageModel(FLAVAPreTrainedModel):
+    config_class = FLAVAImageConfig
+    base_model_prefix = "flava.image_model"
```

Comment:
This model does not have an attribute named `flava.image_model`, so this shouldn't be here.

Reply:
This is so that we can load the `FLAVAImageModel` specifically from `aps/flava-full` checkpoint as well. What is the right way to do this otherwise? There are cases where will want to only load `FLAVAImageModel`.

Reply:
Ah, I understand what you're trying to do. This is a current limitation of the `PreTrainedModel` API, as `base_model_prefix` can only be one value, not several depending on the model. So this is indeed the best way to achieve this.

Reply:
ok for me! 
We have this problem as well for encoder-decoder models in case one wants to load just the decoder from the "full" checkpoint.

Until now, we've build wrappers such as https://github.com/huggingface/transformers/blob/f861504466bf19c607e3c8407be6194a565afc00/src/transformers/models/bart/modeling_bart.py#L1694 to be able to load smaller pretrained checkpoints from the "full" checkpoint.

The somewhat "hacky" way here is ok for me though!

Reply:
I think we also need to find a better solution at the `from_pretrained` level, so in my mind, this hack is temporary.